Assets/Scripts/Dialogue scripts/Dialogue.cs

```diff
@@ -1,197 +1,184 @@
+            else if (_item != null)
+            {
+                if (_item.activeSelf == true) StartCoroutine(RunHiddenParagraphCycle());
+
+                else StartCoroutine(RunParagraphCycle());
+            }
 
             else StartCoroutine(RunParagraphCycle());
         }
 
-        else StartCoroutine(RunParagraphCycle());
-    }
+        public IEnumerator Play(String pageText)
+        {
+            var sb = new StringBuilder();
 
-    public IEnumerator Play(String pageText)
-    {
-        var sb = new StringBuilder();
+            var letters = pageText.ToCharArray();
 
-        var letters = pageText.ToCharArray();
+            foreach (var letter in letters)
+            {
+                sb.Append(letter);
 
-        foreach (var letter in letters)
-        {
-            sb.Append(letter);
+                _pageRender.RenderPageText(sb.ToString());
 
-            _pageRender.RenderPageText(sb.ToString());
+                yield return new WaitForSeconds(0.0375f);
+            }
 
-            yield return new WaitForSeconds(0.0375f);
+       
+            yield return null;
         }
 
-       
-        yield return null;
-    }
+        public IEnumerator RunParagraphCycle()
+        {
+            _dialogueBox.SetActive(true);
 
-    public IEnumerator RunParagraphCycle()
-    {
-        dialogueBox.SetActive(true);
+            _rayCast.SetActive(false);
 
-        rayCast.SetActive(false);
+            int paragraphCounter = 0;
 
-        int paragraphCounter = 0;
+            Coroutine currentRoutine = null;
 
-        Coroutine currentRoutine = null;
+            while (paragraphCounter < _textBox.Length)
+            {
+                if (currentRoutine != null) StopCoroutine(currentRoutine);
 
-        while (paragraphCounter < _textBox.Length)
-        {
-            if (currentRoutine != null) StopCoroutine(currentRoutine);
+                currentRoutine = StartCoroutine(Play(_textBox[paragraphCounter]));
 
-            currentRoutine = StartCoroutine(Play(_textBox[paragraphCounter]));
+                ++paragraphCounter;
 
-            ++paragraphCounter;
+                yield return new WaitForSeconds(0.0375f);
 
-            yield return new WaitForSeconds(0.0375f);
+                while (!Input.GetKeyDown(KeyCode.E))
+                {
+                    yield return null;
+                }
 
-            while (!Input.GetKeyDown(KeyCode.E))
-            {
                 yield return null;
             }
 
-            yield return null;
-        }
-
-        if (currentRoutine != null) StopCoroutine(currentRoutine);
+            if (currentRoutine != null) StopCoroutine(currentRoutine);
 
-        dialogueBox.SetActive(false);
+            _dialogueBox.SetActive(false);
 
-        if (_itemClass >= 1)
-        {
-            joInvScript.AddPage(gameObject);
+            if (_itemClass >= 1)
+            {
+                _joInvScript.AddPage(gameObject);
```

Comment:
the name `_joInvScript` is a bit confusing - maybe consider something a little more explicit?